nix/deployments.nix

```diff
@@ -0,0 +1,42 @@
+# SPDX-FileCopyrightText: 2022-2024 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  self,
+  inputs,
+  ...
+}: let
+  inherit (inputs) deploy-rs;
+
+  mkDeployment = arch: config: hostname: {
+    inherit hostname;
+    remoteBuild = arch != "x86_64-linux";
```

Comment:
This seems unnecessary, and could cause a lot of load to non-x86 machines.

If the closure to deploy is either entirely uploaded to the binary cache, or the machine doing the deployment has a aarch64 builder configured, it should be possible to drop it.

I'd prefer to not set this, and document that there exists a `--remote-build` CLI flag in deploy-rs in cases where you don't have an aarch64-linux builder configured on the machine you deploy from, and it's not in the binary cache.

Reply:
I wanted everything to work by default on x86 without configured remote builders, but in the scope of this project maybe it's fair to assume those exist.